src/datasource.js

```diff
@@ -297,6 +298,11 @@ function (angular, _, dateMath, moment) {
     };
 
     this._druidQuery = function (query) {
+      if (this.queryTimeout != null) {
+        query.context = {};
```

Comment:
should probably check if it already has some non-null value.
`query.context ||=  {}`